extensions/amp-timeago/1.0/amp-timeago.js

```diff
@@ -50,11 +51,35 @@ AmpTimeago['layoutSizeDefined'] = true;
 
 /** @override */
 AmpTimeago['props'] = {
-  'datetime': {attr: 'datetime'},
-  'locale': {attr: 'locale', default: 'en'},
+  'datetime': {
+    attrs: ['datetime', 'timestamp-ms', 'timestamp-seconds', 'offset-seconds'],
+    parseAttrs: parseDateAttrs,
+  },
   'cutoff': {attr: 'cutoff', type: 'number'},
+  'locale': {attr: 'locale'},
 };
 
+/**
+ * @param {!Element} element
+ * @return {?number}
+ * @visibleForTesting
+ */
+export function parseDateAttrs(element) {
+  // TODO(#29246): Is this a coincidence that timeago would have the same format
+  // as date-display? E.g. the format for date-countdown is somewhat different.
+  const datetimeStr = element.getAttribute('datetime');
+  const datetime = parseDate(datetimeStr);
+  const timestampMs = Number(element.getAttribute('timestamp-ms'));
+  const timestampSeconds =
+    Number(element.getAttribute('timestamp-seconds')) * 1000;
+  const offsetSeconds = Number(element.getAttribute('offset-seconds')) * 1000;
+
+  const epoch = datetime || timestampMs || timestampSeconds;
```

Comment:
Nit: We should do these the accesses inline, so that DOM access are not done unnecessarily.